src/components/modals/Dialog.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import FocusTrap from 'focus-trap-react';
+import noScroll from 'no-scroll';
+import rafSchd from 'raf-schd';
+import DenyIcon from '../../../lib/icons/react/DenyIcon';
+import Flex from '../layout/Flex';
+import FlexItem from '../layout/FlexItem';
+import Section from '../layout/Section';
+
+export const isElementOverflowing = ({ current }) => {
+  // Checking for current first is safer just in case,
+  // and will also fail gracefully in tests
+  if (current) {
+    return current.scrollHeight > current.clientHeight;
+  }
+  return false;
+};
+
+const Dialog = (props) => {
+  const contentEl = useRef(null);
+
+  const handleKeyDown = (e) => {
+    if (props.onDismiss && e.key === 'Escape') {
+      props.onDismiss();
+    }
+  };
+
+  const [isOverflowing, setIsOverflowing] = useState(false);
+
+  const handleResize = () => {
+    rafSchd(setIsOverflowing(isElementOverflowing(contentEl)));
+  };
+
+  useLayoutEffect(() => {
```

Comment:
not a blocker, but i think this could use regular ol' `useEffect` because the resize listener doesn't need to do any DOM API manipulation